src/globus_sdk/utils.py

```diff
@@ -147,28 +147,70 @@ class PayloadWrapper(PayloadWrapperBase):
     # internal helpers for setting non-null values
     #
 
+    # Some payloads use None to represent "no value" and others use the MISSING sentinel
+    _MISSING_SENTINEL: None | MissingType = None
```

Comment:
I would prefer that we not introduce this as a mechanism for skipping over `MISSING` values when they are passed for a PayloadWrapper. In the normal case, any class which accepts `MISSING` can do so like so:
```python
def __init__(self, foo: int | MissingType = MISSING):
    super().__init__()
    self["foo"] = foo
```

Having a path for skipping MISSING therefore seems inappropriate, since one of the main goals in introducing it was to allow a caller to pass the value forward and rely on the request-time serialization code to remove it.

I don't believe it's necessary to alter the `_set_value` and `_set_opt*` helpers in order to handle `MISSING` fields, and I'd instead prefer that we simply stop using those helpers for new classes as appropriate.

I recognize that there are callbacks being invoked here, but I think we can generally avoid them as well. I'll need to work up a a PR against this PR to make the recommendation which I think is appropriate. It's a little too much for the GitHub suggestion UI.